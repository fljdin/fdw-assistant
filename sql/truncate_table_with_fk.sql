CREATE SCHEMA source;

-- "parent" and "child" are involved in a foreign key
CREATE TABLE source.parent (id int);
CREATE TABLE public.parent (id int primary key);

CREATE TABLE source.child (id int, parent_id int);
CREATE TABLE public.child (id int primary key, parent_id int not null references public.parent(id));

INSERT INTO config (source, target, pkey, trunc) VALUES
-- a truncate should failed if target is involved in a foreign key
    ('source.parent', 'public.parent', 'id', true);

SELECT invocation FROM plan('{public.parent}');

-- copy(1) should fail because "parent" is referenced by "child"
CALL copy(1);

-- job 1 should have a failed state
SELECT stage_id, target, rows, state
  FROM report WHERE stage_id = 1 ORDER BY target;
