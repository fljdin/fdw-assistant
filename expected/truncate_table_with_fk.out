CREATE SCHEMA source;
CREATE SCHEMA
-- "parent" and "child" are involved in a foreign key
CREATE TABLE source.parent (id int);
CREATE TABLE
CREATE TABLE public.parent (id int primary key);
CREATE TABLE
CREATE TABLE source.child (id int, parent_id int);
CREATE TABLE
CREATE TABLE public.child (id int primary key, parent_id int not null references public.parent(id));
CREATE TABLE
INSERT INTO config (source, target, pkey, trunc) VALUES
-- a truncate should failed if target is involved in a foreign key
    ('source.parent', 'public.parent', 'id', true);
INSERT 0 1
SELECT invocation FROM plan('{public.parent}');
       invocation        
-------------------------
 CALL assistant.copy(1);
(1 row)

-- copy(1) should fail because "parent" is referenced by "child"
CALL copy(1);
NOTICE:  Executing: TRUNCATE public.parent
ERROR:  cannot truncate a table referenced in a foreign key constraint
CONTEXT:  SQL statement "TRUNCATE public.parent"
